receiver/rabbitmq.go

```diff
@@ -0,0 +1,111 @@
+/*
+ * skogul, rabbitmq-receiver
+ *
+ * Copyright (c) 2023 Telenor Norge AS
+ * Author(s):
+ *  - Kamil Oracz <kamil.oracz@telenor.no>
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+ * 02110-1301  USA
+ */
+
+package receiver
+
+import (
+	"fmt"
+
+	amqp "github.com/rabbitmq/amqp091-go"
+	"github.com/telenornms/skogul"
+)
+
+type Rabbitmq struct {
+	Username string             `doc:"Username for rabbitmq instance"`
+	Password string             `doc:"Password for rabbitmq instance"`
```

Comment:
This could use the `Secret` type (instead of string) to avoid being (accidentally) exposed in logs etc: https://github.com/telenornms/skogul/blob/primary/common.go#L348